ocaml/xapi/updateinfo.ml

```diff
@@ -605,94 +605,103 @@ module UpdateInfo = struct
+                              | Some date -> (
+                                try
+                                  Xapi_stdext_date.Date.of_string
+                                    (Scanf.sscanf date
+                                       "%04d-%02d-%02d %02d:%02d:%02d"
+                                       (fun y mon d h m s ->
+                                         Printf.sprintf
+                                           "%04i%02i%02iT%02i:%02i:%02iZ" y mon
+                                           d h m s
+                                     )
+                                    )
+                                with e ->
+                                  (* The error should not block update. Ingore it
+                                     and set "issued" as epoch. *)
+                                  warn "%s" (ExnHelper.string_of_exn e) ;
+                                  Xapi_stdext_date.Date.epoch
+                              )
+                              | None ->
+                                  Xapi_stdext_date.Date.epoch
+                            in
+                            {acc with issued}
+                        | Xml.Element ("severity", _, [Xml.PCData v]) -> (
+                          try {acc with severity= Severity.of_string v}
+                          with e ->
+                            (* The error should not block update. Ingore it. *)
+                            warn "%s" (ExnHelper.string_of_exn e) ;
+                            acc
+                        )
+                        | Xml.Element ("title", _, [Xml.PCData v]) ->
+                            {acc with title= v}
+                        | _ ->
+                            acc
                       )
-                      | Xml.Element ("title", _, [Xml.PCData v]) ->
-                          {acc with title= v}
-                      | _ ->
-                          acc
-                    )
-                    {default with update_type= ty}
-                    update_nodes
-                  |> assert_valid_updateinfo
-                in
-                debug "updateinfo: %s" (to_string ui) ;
-                Some ui
-            | _ ->
-                None
-          )
-          children
-        |> assert_no_dup_update_id
-        |> List.map (fun updateinfo -> (updateinfo.id, updateinfo))
+                      {default with update_type= ty}
+                      update_nodes
+                    |> assert_valid_updateinfo
+                  in
+                  debug "updateinfo: %s" (to_string ui) ;
+                  Some ui
+              | _ ->
+                  None
+            )
+            children
+          |> assert_no_dup_update_id
+          |> List.map (fun updateinfo -> (updateinfo.id, updateinfo))
+        in
+        match (api_ver, uis) with
+        | Some _v, [] ->
+            raise Api_errors.(Server_error (invalid_updateinfo_xml, []))
```

Comment:
It would be good to log that offending XML code - how are we going to debug this otherwise?